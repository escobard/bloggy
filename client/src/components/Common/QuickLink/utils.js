export const aNumber = (answers) => {
		let array = Object.values(answers);
		return array.length;
};