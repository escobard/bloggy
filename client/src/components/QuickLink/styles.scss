@import "../../styles/variables";
@import "../../styles/mixins";

.dashboard-cta {
	.card {
		min-height: $space-xxmega;
		background-color: $primary-dark !important;
		color: $white !important;
		border-radius: $space-small !important;
		margin: 0 !important;
		padding: $space-xlrg 0;
		// this needs to be fixed, and refatored into a mixin
		&:hover {
			-webkit-box-shadow: 0px 0px 5px 5px rgba(2, 117, 216, 1);
			-moz-box-shadow: 0px 0px 5px 5px rgba(2, 117, 216, 1);
			box-shadow: 0px 0px 3px 2px rgb(0, 24, 97) !important;
		}
		svg {
			@include svg-icon();
		}
		h4 {
			text-align: center;
			text-transform: capitalize;
			margin-top: $space-lrg;
		}
		img {
			width: 100%;
			float: none;
			margin: auto;
		}
	}
}