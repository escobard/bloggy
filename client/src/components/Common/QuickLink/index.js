import React, { Component } from "react";
import PropTypes from "prop-types";
import { RaisedButton, Card } from "material-ui";
import { Link } from "react-router-dom";
import DashIcon from "material-ui/svg-icons/action/dashboard";

import styles from "./styles.scss";

// add images and finish layout of 2x2 for each cta
// <img src="" alt={`${route} icon`} /> --- code for image when icon is replaced
export default class QuickLink extends Component {
	render() {
		let { img, route } = this.props.route;
		return (
			<div className="dashboard-cta col-md-4">
				<Link to={`/dashboard/${route}`}>
					<Card className="card">
						<DashIcon />
						<h4>{route}</h4>
					</Card>
				</Link>
			</div>
		);
	}
}

QuickLink.defaultProps = {
	route: {
		img: "test",
		route: "test"
	}
};

QuickLink.propTypes = {
	routeObject: PropTypes.object
};