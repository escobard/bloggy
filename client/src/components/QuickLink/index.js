import React, { Component } from "react";
import PropTypes from "prop-types";
import { RaisedButton, Card } from "material-ui";
import { Link } from "react-router-dom";
import DashIcon from "material-ui/svg-icons/action/dashboard";

import styles from "./styles.scss";

// add images and finish layout of 2x2 for each cta
export default class QuickLink extends Component {
	render() {
		let { img, route } = this.props.route;
		return (
			<div className="dashboard-cta col-md-3">
				<Link to={`/dashboard/${route}`}>
					<Card className="card">
						<img src="" alt={`${route} icon`} />
						<h4>{route}</h4>
					</Card>
				</Link>
			</div>
		);
	}
}

QuickLink.defaultProps = {
	route: {
		img: "test",
		route: "test"
	}
};

QuickLink.propTypes = {
	routeObject: PropTypes.object
};