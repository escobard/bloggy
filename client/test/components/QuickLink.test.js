import React, { Component } from 'react'
import QuickLink from '../../src/components/QuickLink';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('QuickLink snapshot', () => {
	const initialState = {output:100}
	const mockStore = configureStore()
	let store,container

    beforeEach(()=>{
        store = mockStore(initialState)
        container = shallow(<QuickLink /> )  
    })

    it('+++ render the QuickLink component', () => {
       expect(toJson(container)).toMatchSnapshot()
    });

});