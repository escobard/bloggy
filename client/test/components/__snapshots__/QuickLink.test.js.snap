// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLink snapshot +++ render the QuickLink component 1`] = `
<div
  className="dashboard-cta col-md-3"
>
  <Link
    replace={false}
    to="/dashboard/test"
  >
    <Card
      className="card"
      expandable={false}
      expanded={null}
      initiallyExpanded={false}
    >
      <img
        alt="test icon"
        src=""
      />
      <h4>
        test
      </h4>
    </Card>
  </Link>
</div>
`;
